oak_idl/src/lib.rs

```diff
@@ -20,120 +20,39 @@ extern crate alloc;
 
 use alloc::vec::Vec;
 use core::fmt::Debug;
-use serde::{Deserialize, Serialize};
 
 pub mod utils;
 
 #[derive(Debug)]
-pub enum ClientError {
+pub enum Error {
+    /// The request message could not be deserialized correctly.
     InvalidRequest,
+    /// The response message could not be serialized correctly.
```

Comment:
Should those comments be swapped? Surely the request would get serialised, and the response is deserialized

Reply:
I think they are the right way round, but let me double check by writing it down.

The generated client takes a `&[u8]` (which must have been serialized by the caller), and tries to deserialize it to make sure it corresponds to the correct type. If that fails, then the client returns `InvalidRequest` to the caller directly.

When the server receives the request, it also tries to deserialize it, and returns `InvalidRequest` to the client in that case.

The generated client also deserializes the response from the server, and if that fails it returns `InvalidResponse`. As I'm typing this, I realise that indeed the comment on `InvalidResponse` was incorrect!

So perhaps actually both comments should say "deserialized", since as a matter of fact the serialization to flatbuffer is done entirely by the caller (never by the generated code). Interesting.

Reply:
(fixed the comment now, thanks!)

Reply:
Thanks for updating, that makes a lot more sense. What's the rationale behind the Client accepting a byte slice (and deserializing it to verify) as opposed to instance of the `request_type`?